ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13477,6 +13480,23 @@ private boolean hasConstraints(final List<FieldSchema> partCols, final List<SQLD
     }
     return false;
   }
+
+  boolean doesSupportSchemaInference(String fileFormat) throws SemanticException {
```

Comment:
Pls add a comment.  Also, this could be a public static utility method that takes fileFormat and the conf parameters.